File: app/src/main/java/com/sorcerer/sorcery/iconpack/utils/LocaleUtil.java
package com.sorcerer.sorcery.iconpack.utils;

import android.content.Context;
import android.content.res.Configuration;
import android.content.res.Resources;
import android.util.DisplayMetrics;

import com.sorcerer.sorcery.iconpack.R;

import java.util.Locale;

/**
 * @description: *
 * @author: Sorcerer
 * *
 * @date: 2017/2/5
 */

public class LocaleUtil {
    public static boolean isChinese(Context context) {
        return context.getResources().getString(R.string.language).equals("zh");
    }

    public void forceChinese(Context context) {
        Resources res = context.getResources();
        DisplayMetrics dm = res.getDisplayMetrics();
        Configuration conf = res.getConfiguration();
        conf.locale = new Locale("zh");
        res.updateConfiguration(conf, dm);
    }
}

File: app/src/main/java/com/sorcerer/sorcery/iconpack/utils/LocaleUtil.kt
package com.sorcerer.sorcery.iconpack.utils

import android.content.Context
import com.sorcerer.sorcery.iconpack.R
import java.util.*

/**
 * @description:
 * *
 * @author: Sorcerer
 * *
 * @date: 2017/2/5
 */

object LocaleUtil {
    fun isChinese(context: Context): Boolean {
        return context.resources.getString(R.string.language) == "zh"
    }

    @Suppress("DEPRECATION")
    fun forceChinese(context: Context) {
        val res = context.resources
        val dm = res.displayMetrics
        val conf = res.configuration
        conf.locale = Locale("zh")
        res.updateConfiguration(conf, dm)
    }
}
